### PR TITLE
embuilder support for building LTO versions of libraries, '--lto'

### LINKED ISSUE #8128: embuilder: Building bitcode and object libraries
Right now with the LLVM wasm backend, embuilder will build object file libraries for libc, libc++ etc. It came up in https://github.com/emscripten-core/emscripten/pull/7929 that we should probably build the bitcode versions too, for people that want to do bitcode LTO.

Automatically building them would basically double the time it takes to run embuilder. May be worth it, as it's the simplest thing.

Another option is to add an "lto" option for embuilder, that builds the LTO versions. That would give embuilder users more control (maybe some don't need LTO libraries built).

### PR DESCRIPTION
**Pull request by kripken**
fixes #8128

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -351,6 +351,11 @@ jobs:
           command: |
             EMCC_CORES=4 python3 ~/emscripten/embuilder.py build ALL
             python3 ~/emscripten/tests/runner.py test_hello_world
+      - run:
+          name: embuilder build for lto
+          command: |
+            EMCC_CORES=4 python3 ~/emscripten/embuilder.py build libc++ --lto
```
Comment (sbc100) [inline .circleci/config.yml]: Why not `build ALL`

File: embuilder.py [hunk 2]
```diff
@@ -150,7 +152,7 @@ def build_port(port_name, lib_name, params):
 
 
 def static_library_name(name):
-  if shared.Settings.WASM_BACKEND:
+  if shared.Settings.WASM_BACKEND and shared.Settings.WASM_OBJECT_FILES:
```
Comment (sbc100) [inline embuilder.py]: I think the extension should always be .a here, even in the LTO case.

Regardless of whether, the files inside are `.bc` or `.o` I think archives should always be called `.a`.  The exception is for fastcomp that doesn't use libraries at all but llvm-link'd bitcode.

File: tests/test_sanity.py [hunk 3]
```diff
@@ -973,3 +973,15 @@ def prep():
           if not side_module:
             assert os.path.exists('a.out.js')
             self.assertContained('hello, world!', run_js('a.out.js'))
+
+  def test_embuilder(self):
+    restore_and_set_up()
```
Comment (sbc100) [inline tests/test_sanity.py]: How about just skip this test if `is_wasm_backend()`?

File: .circleci/config.yml (hunk 1 above)
Comment (kripken) [reply .circleci/config.yml]: `ALL` takes around 5 minutes, and we only need libc++ for the tests we currently have. Just trying to save time on the bots.

File: embuilder.py (hunk 2 above)
Comment (kripken) [reply embuilder.py]: I don't feel strongly either way, but this PR is just matching the current behavior defined here: https://github.com/emscripten-core/emscripten/blob/incoming/tools/system_libs.py#L616

File: tests/test_sanity.py (hunk 3 above)
Comment (kripken) [reply tests/test_sanity.py]: Ok. I'll rename the test too, then, to `test_embuilder_wasm_backend.`

**Comment by kripken** [comment]
Thanks, if you want to do argparse that sounds good.

**Pull request merged**